spring-integration-file/src/main/java/org/springframework/integration/file/dsl/RemoteFileOutboundGatewaySpec.java

```diff
@@ -358,6 +359,48 @@ public S fileExistsMode(FileExistsMode fileExistsMode) {
 		return _this();
 	}
 
+	/**
+	 * Specify a SpEL expression to determine the action to take when files already exist.
+	 * Expression evaluation should return a {@link FileExistsMode} or a String representation.
+	 * Used for GET and MGET operations when the file already exists locally,
+	 * or PUT and MPUT when the file exists on the remote system.
+	 * @param fileExistsModeExpression a SpEL expression to evaluate the file exists mode
+	 * @return the Spec.
+	 * @since 6.5
+	 */
+	public S fileExistsModeExpression(Expression fileExistsModeExpression) {
+		this.target.setFileExistsModeExpression(fileExistsModeExpression);
+		return _this();
+	}
+
+	/**
+	 * Specify a SpEL expression to determine the action to take when files already exist.
+	 * Expression evaluation should return a {@link FileExistsMode} or a String representation.
+	 * Used for GET and MGET operations when the file already exists locally,
+	 * or PUT and MPUT when the file exists on the remote system.
+	 * @param fileExistsModeExpression the String in SpEL syntax.
+	 * @return the Spec.
+	 * @since 6.5
+	 */
+	public S fileExistsModeExpression(String fileExistsModeExpression) {
+		this.target.setFileExistsModeExpressionString(fileExistsModeExpression);
+		return _this();
+	}
+
+	/**
+	 * Specify a {@link Function} to determine the action to take when files already exist.
+	 * Expression evaluation should return a {@link FileExistsMode} or a String representation.
+	 * Used for GET and MGET operations when the file already exists locally,
+	 * or PUT and MPUT when the file exists on the remote system.
+	 * @param fileExistsModeFunction the {@link Function} to use.
+	 * @param <P> the expected payload type.
+	 * @return the Spec.
+	 * @since 6.5
+	 */
+	public <P> S fileExistsModeFunction(Function<Message<P>, String> fileExistsModeFunction) {
```

Comment:
`String` is not correct here since we also expect a `FileExistsMode`.
Therefore just `Object`

Reply:
Fixed the `Function` parameter type to `Object` as suggested. I also noticed and fixed an incorrect method call in the `fileExistsModeFunction()` implementation (it was incorrectly calling `remoteDirectoryExpression()` instead of `fileExistsModeExpression()`).